CVLByExample/LastReverted/RevertingConditionsForgotWithRevert.spec

```diff
@@ -0,0 +1,37 @@
+methods {
+    function balanceOf(address) external returns (uint256) envfree;
+    function transfer(address,uint256) external returns (bool);
+    function approve(address,uint256) external returns (bool);
+}
+
+rule transferRevertingConditions {
+    env e;
+    address recipient;
+    uint256 amount;
+
+    require e.msg.value == 0;
+    require amount + balanceOf(recipient) <= max_uint256;
+    
+    bool senderIsZero = e.msg.sender == 0;
+    bool recipientIsZero = recipient == 0;
+    bool notEnoughFunds = balanceOf(e.msg.sender) < amount;
+    
+    transfer(e, recipient, amount);
+
+    assert lastReverted <=> senderIsZero || recipientIsZero || notEnoughFunds;
```

Comment:
lets make this more interesting and common:

```
transfer@withrevert(e, recipient, amount);

assert balanceOf(e.msg.sender) > balanceBefore - amount =>  lastReverted

Reply:
Nice one Nurit, this doesn't work: https://prover.certora.com/output/15800/4e4bcbd0c5384d0ab0e7109748e5cb1f?anonymousKey=7f7323ca7416e69a40bafc323b10106d5682356a